components/List/examples/supplemental/MDCListItemCell.h

```diff
@@ -0,0 +1,104 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "MDCListBaseCell.h"
+
+/**
+ A basic Material List Cell with labels and imageviews.
+ */
+@interface MDCListItemCell : MDCListBaseCell
+
+/**
+ Sets the cell width for the self-sizing cell.
+ Note: The self-sizing is only applied to the height, and the width is set using this method.
+ */
+@property (nonatomic, assign) CGFloat cellWidth;
+
+/**
+ A convenience accessor for the title text.
+ */
+@property (nonatomic, copy, nullable) NSString *titleText;
+
+/**
+ A convenience accessor for the title text color.
+ */
+@property (nonatomic, strong, nullable) UIColor *titleLabelTextColor;
+
+/**
+ A convenience accessor for the title text font.
+ */
+@property (nonatomic, strong, nullable) UIFont *titleLabelFont;
+
+/**
+ A convenience accessor for the title text font.
+ */
+@property (nonatomic, assign) NSTextAlignment titleLabelTextAlignment;
+
+/**
+ A convenience accessor for the detail text.
+ */
+@property (nonatomic, copy, nullable) NSString *detailText;
+
+/**
+ A convenience accessor for the detail text color.
+ */
+@property (nonatomic, strong, nullable) UIColor *detailLabelTextColor;
+
+/**
+ A convenience accessor for the detail text font.
+ */
+@property (nonatomic, strong, nullable) UIFont *detailLabelFont;
+
+/**
+ A convenience accessor for the detail text font.
+ */
+@property (nonatomic, assign) NSTextAlignment detailLabelTextAlignment;
+
+/**
+ The UIImageview used to display the leading image.
+ */
+@property (strong, nonatomic, nonnull, readonly) UIImageView *leadingImageView;
+
+/**
+ A convenience accessor for the leading image.
+ */
+@property (strong, nonatomic, nullable) UIImage *leadingImage;
+
+/**
+ A convenience accessor for the leading image corner radius.
+ */
+@property (nonatomic, assign) UIImage *leadingImageCornerRadius;
+
+/**
+ A convenience accessor for the trailing image.
+ */
+@property (strong, nonatomic, nullable) UIImage *trailingImage;
+
+/**
+ A convenience accessor for the trailing image corner radius.
+ */
+@property (nonatomic, assign) UIImage *trailingImageCornerRadius;
```

Comment:
Why does an image need a corner radius? Can't clients just set a rounded rectangular image?

Reply:
This was a request from @ianegordon. He asked that I get rid of the publicly exposed views and replace them with the "pass through" accessors you see here. This was one he mentioned.

You also noticed [here](https://github.com/material-components/material-components-ios/pull/4408#discussion_r197149877) that a UIImageView is still exposed. That was a mistake..

This question of whether or not to expose the views is something that we should all discuss more, because my instinct was actually to expose readonly views rather than have a boatload of pass through properties. I think Ian actually began to reconsider his earlier stance once he realized we'd have to expose pass through properties for all the accessibility related view properties as well. At some point it just becomes too much...

Reply:
Apologies if I was not clear.  Based on our earlier discussion re: customizability and accessibility, I believe we will need to expose all of UIViews (the two imageviews and the two labels) instead of just the passthrough properties.

Reply:
Fair enough, but where does the corner radius come into play? I don't see any such property on `UIImageView`.  Is this being added in iOS 12 and I just missed the update?

Reply:
It's on the `CALayer`, of course... I think the idea is just that rounding an image is something people would want to often.

I know the three of us talked about this offline yesterday--I think I'm still in favor of exposing the UIViews 😅